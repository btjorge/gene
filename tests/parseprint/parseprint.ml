open Genelib

let prefix = "../progs/"

(* In theory all the progs can be parsed *)

let progs = List.sort compare (Array.to_list (Sys.readdir prefix))

let parseprint file =
  let ic = open_in (prefix ^ file) in
  let lexbuf = Lexing.from_channel ic in

  let p = Parser.prog Lexer.token lexbuf in

  Printer.print_prog p

let () =
  List.iter parseprint progs
