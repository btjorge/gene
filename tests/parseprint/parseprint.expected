max (x, y) {
  if (x <= y) {
    return y
  } else {
    return x
  }
}

abs (x) {
  return max (x, (0 - x))
}

test () {
  assertEquals (abs (5), 5)
}
test () {
  assertEquals (max (5, 2), 5)
}
max (x, y) {
  return 5
}

test () {
  assertEquals (max (5, 2), 5)
}
max (x, y) {
  return x
}

test () {
  assertEquals (max (5, 2), 5)
}
test () {
  assertEquals (5, 5)
}
max (x, y) {
  return y
}

test () {
  assertEquals (max (5, 2), 5)
}
max (x, y) {
  if (x <= y) {
    return x
  } else {
    return y
  }
}

test () {
  assertEquals (max (5, 2), 5)
}
max (x, y) {
  if (x <= y) {
    return y
  } else {
    return x
  }
}

test () {
  assertEquals (max (5, 2), 5)
}
test (x, y, z) {
  if (x <= y) {
    assertEquals (y, z)
  } else {
    assertEquals (x, z)
  }
}
max (x, y) {
  return 5
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6)
}
max (x, y) {
  return x
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6)
}
max (x, y) {
  if (x <= 2) {
    return 5
  } else {
    return x
  }
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6);
  assertEquals (max (2, 5), 5)
}
max (x, y) {
  if (x <= 2) {
    return y
  } else {
    return x
  }
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6);
  assertEquals (max (2, 5), 5)
}
max (x, y) {
  if (x <= y) {
    return x
  } else {
    return y
  }
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6);
  assertEquals (max (2, 5), 5)
}
max (x, y) {
  if (x <= y) {
    return y
  } else {
    return x
  }
}

test () {
  assertEquals (max (5, 2), 5);
  assertEquals (max (6, 2), 6);
  assertEquals (max (2, 5), 5)
}
max (x, y) {
  if (x <= y) {
    return y
  } else {
    return x
  }
}

max3 (x, y, z) {
  return max (max (x, y), z)
}

test () {
  assertEquals (max3 (1, 3, 2), 3)
}
test () {
  x := 2;
  y := 3;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, 3)
}
test () {
  x := 2;
  y := 3;
  if (x <= y) {
    z := y
  } else {
    z := x
  };
  assertEquals (z, y)
}
